src/StdLib/StdForwardList.jl

```diff
@@ -4,4 +4,30 @@ Base.isempty(v::StdForwardList) = flist_isempty(v)
 Base.first(v::StdForwardList) = flist_front(v)
 Base.empty!(v::StdForwardList) = (flist_empty!(v); v)
 Base.pushfirst!(v::StdForwardList, x) = (flist_push_front!(v, x); v)
-Base.popfirst!(v::StdForwardList) = (flist_pop_front!(v); v)
\ No newline at end of file
+Base.popfirst!(v::StdForwardList) = (flist_pop_front!(v); v)
+
+Base.:(==)(a::StdForwardListIterator, b::StdForwardListIterator) = iterator_is_equal(a, b)
+_forward_list_iteration_tuple(v::StdForwardList, state::StdForwardListIterator) = (state == iteratorend(v)) ? nothing : (iterator_value(state), state)
+Base.iterate(v::StdForwardList) = _forward_list_iteration_tuple(v, iteratorbegin(v))
+Base.iterate(v::StdForwardList, state::StdForwardListIterator) = (state != iteratorend(v)) ? _forward_list_iteration_tuple(v, iterator_next(state)) : nothing
+
+function Base.show(io::IO, ::MIME"text/plain", container::StdForwardList)
+    print(io, "StdForwardList[")
+
```

Comment:
I would rather go for something like:

```
StdForwardList{<element type>} with X elements:
  2
  3
  1
```
similar to the way Julia `Set`s are printed. The [] syntax risks confusion with arrays.

Reply:
For a forward list, printing the length will require iterating over the entire list, so I have omitted the number of elements and gone for this:

```
StdForwardList{Int64}:
  2
  3
  1
  ```